### PR TITLE
Issue 10 expose list of artificats

### PR DESCRIPTION
**Pull request by GiriDandu**
Add testing based on multiple artifacts based on review comments

### CODE REVIEW

File: features/api/artifact/reading.feature [hunk 1]
```diff
@@ -51,38 +51,66 @@ Feature: Artifact READ APIs
...
     When I GET "/api/v1/artifacts/by-module/com.example.cucumber:cucumber-artifact?pageNumber=1&perPageSize=5"
     Then the response should be 404
 
+  @error
   Scenario: Fetching an artifact by the (group, name) tuple with invalid data type for pageSize
 
     When I GET "/api/v1/artifacts/by-module/com.example.cucumber:cucumber-artifact?pageNumber=-1&perPageSize=0xdeadbeef"
     Then the response should be 400
 
+@wip
```
Comment (rtyler) [inline features/api/artifact/reading.feature]: Based on some discussion we had previously (can't find the link now) I was under the impression that this API with a zero page size and a zero per-page, we'd return an empty list"

File: src/test/groovy/deploydb/models/ArtifactSpec.groovy [hunk 2]
```diff
@@ -3,6 +3,16 @@ package deploydb.models
 import spock.lang.*
 
 class ArtifactSpec extends Specification {
+
+    private Artifact artifact
+
+    def setup() {
```
Comment (rtyler) [inline src/test/groovy/deploydb/models/ArtifactSpec.groovy]: Why is this here? It's not being used by anything

File: src/test/groovy/deploydb/models/ArtifactSpec.groovy [hunk 3]
```diff
@@ -14,6 +24,15 @@ class ArtifactSpec extends Specification {
 
 class ArtifactWithArgsSpec extends Specification {
 
+    private Artifact artifact
+
+    def setup() {
+        artifact = new Artifact('spock.group',
```
Comment (rtyler) [inline src/test/groovy/deploydb/models/ArtifactSpec.groovy]: Same comment as above, I don't understand why this setup() method is here

File: features/api/artifact/reading.feature (hunk 1 above)
Comment (GiriDandu) [inline features/api/artifact/reading.feature]: Agreed, Thats why I move this test to @wip and I am tracking this with another issue #32.

File: src/test/groovy/deploydb/models/ArtifactSpec.groovy (hunk 3 above)
Comment (GiriDandu) [inline src/test/groovy/deploydb/models/ArtifactSpec.groovy]: Put back the change that @maheshkelkar made.  Fix with latest commit
